src/main/java/com/azure/cosmos/kafka/connect/sink/CosmosDBSinkTask.java

```diff
@@ -90,11 +90,7 @@ public void put(Collection<SinkRecord> records) {
     }
 
     private void addItemToContainer(CosmosContainer container, Object recordValue) {
-        if (config.getUseUpsert().equalsIgnoreCase(CosmosDBConfig.BooleanValues.TRUE.toString())) {
-            container.upsertItem(recordValue);
-        } else {
-            container.createItem(recordValue);
-        }
+        container.upsertItem(recordValue);

```

Comment:
@kushagraThapar what are the perf implications of always doing upsertItem instead of insertItem?

Reply:
Upsert internally does a read first, so there will be some performance degradation and more RU cost for each operation. However, this will make sure the correctness of the SDK in every case. That is basically the trade off we have decided at this point of time. 
Once we have covered all the corner cases of auto-generation of ids, we can then start using insert and expose the configuration back again to the customers.

Reply:
@moderakh - thoughts ?

Reply:
as @ncliang explained i believe this is an acceptable compromise short term. once we implement finer grained control over the checkpointing and have DLQ implemented properly we can look to reintroduce this. so i am ok with this as an interim.

Reply:
upsert is the default on spark. and the perf difference is negligible. As the create story is not compatible with retry we should remove this for now.